Mathlib/Topology/Homeomorph/Lemmas.lean

```diff
@@ -28,13 +28,30 @@ variable [TopologicalSpace X] [TopologicalSpace Y] [TopologicalSpace W] [Topolog
   {X' Y' : Type*} [TopologicalSpace X'] [TopologicalSpace Y']
 
 /-- Homeomorphism given an embedding. -/
-noncomputable def ofIsEmbedding (f : X → Y) (hf : IsEmbedding f) : X ≃ₜ Set.range f where
-  continuous_toFun := hf.continuous.subtype_mk _
-  continuous_invFun := hf.continuous_iff.2 <| by simp [continuous_subtype_val]
-  toEquiv := Equiv.ofInjective f hf.injective
+@[simps! apply_coe]
+noncomputable def _root_.Topology.IsEmbedding.toHomeomorph {f : X → Y} (hf : IsEmbedding f) :
+    X ≃ₜ Set.range f :=
+  Equiv.ofInjective f hf.injective |>.toHomeomorphOfIsInducing <|
+    IsInducing.subtypeVal.of_comp_iff.mp hf.toIsInducing
+
+@[deprecated (since := "2025-04-16")]
+alias ofIsEmbedding := IsEmbedding.toHomeomorph
+
+@[deprecated (since := "2024-10-26")]
+alias ofEmbedding := IsEmbedding.toHomeomorph
+
+/-- A surjective embedding is a homeomorphism. -/
+@[simps! apply]
+noncomputable def _root_.Topology.IsEmbedding.toHomeomorphOfSurjective {f : X → Y}
+    (hf : IsEmbedding f) (hsurj : Function.Surjective f) : X ≃ₜ Y :=
+  Equiv.ofBijective f ⟨hf.injective, hsurj⟩ |>.toHomeomorphOfIsInducing hf.toIsInducing
+
+@[deprecated (since := "2025-04-16")]
+alias _root_.Topology.IsEmbedding.toHomeomorph_of_surjective :=
+  IsEmbedding.toHomeomorphOfSurjective
 
 @[deprecated (since := "2024-10-26")]
-alias ofEmbedding := ofIsEmbedding
+alias _root_.Embedding.toHomeomeomorph_of_surjective := IsEmbedding.toHomeomorphOfSurjective
```

Comment:
Note this isn't actually me changing a deprecation: the existing deprecation is a few lines up, and this one comes from the other file.